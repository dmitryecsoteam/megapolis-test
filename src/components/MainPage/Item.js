import React from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';

const Item = ({ id, title, deleteItem }) => (
    <div className='item__container'>
        <div className='item__id'>{id}</div>
        <div className='item__title'>{title}</div>
        <div className='item__actions-container'>
            <Link className='item__action' to={`/items/${id}`} >EDIT</Link>
            <button className='item__action' onClick={deleteItem} >DELETE</button>
        </div>
    </div>
);

Item.propTypes = {
    id: PropTypes.number.isRequired, 
    title: PropTypes.string.isRequired,
    deleteItem: PropTypes.func.isRequired
}

export default Item;