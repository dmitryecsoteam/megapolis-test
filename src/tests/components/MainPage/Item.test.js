import React from 'react';
import { shallow } from 'enzyme';
import Item from '../../../components/MainPage/Item';

let wrapper;
const deleteItem = jest.fn();
const id = 1, title = 'test';

beforeEach(() => {
    wrapper = shallow(<Item id={id} title={title} deleteItem={deleteItem} />);
});

test('should render Item component', () => {
    expect(wrapper).toMatchSnapshot();
});

test('should call deleteItem with correct id on button click', () => {
    wrapper.find('button').simulate('click');
    expect(deleteItem).toHaveBeenCalledTimes(1);
    expect(deleteItem).toHaveBeenCalledWith(id);
});