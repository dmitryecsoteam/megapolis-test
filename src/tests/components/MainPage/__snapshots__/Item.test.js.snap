// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render Item component 1`] = `
<div
  className="item__container"
>
  <div
    className="item__id item__cell"
  >
    Задача №
    1
  </div>
  <div
    className="item__title item__cell"
  >
    test
  </div>
  <div
    className="item__actions-container item__cell"
  >
    <Link
      className="item__action item__action--link"
      to="/items/1"
    >
      <img
        alt="Edit"
        className="item__icon"
        src="edit.svg"
      />
    </Link>
    <button
      className="item__action item__action--button"
      onClick={[Function]}
    >
      <img
        alt="Delete"
        className="item__icon"
        src="delete.svg"
      />
    </button>
  </div>
</div>
`;
